
syntax = "proto3";

option objc_class_prefix = "HLW";

package helloworld;

service AtomicSwap {
  rpc ProcessInitiate(Initiate) returns (InitiateReply) {}
  rpc ProcessInitiateSwap(InitiateSwap) returns (AcceptSwap) {}
  rpc ProcessRedeemed(InitiatorRedeemFinished) returns (ParticipantRedeemFinished) {}
}

message Initiate {
  double init_amount = 1;
  double part_amount = 2;
  string init_addr = 3;
}

message InitiateReply {
  string part_addr = 1;
}

message InitiateSwap {
  string init_ctc_hex = 1;
  string init_ctc_tx_hex = 2;
  string init_ctc_redeem_addr = 3;
  string hashed_secret = 4;
}

message AcceptSwap {
  string part_ctc_redeem_addr = 1;
}

message InitiatorRedeemFinished {
  string secret = 1;
}

message ParticipantRedeemFinished {
  bool finished = 1;
}